File: ISP/Kata_ Shops Multirepository/SolucionPhp/Shop/Notifier/ShopSmsNotifier.php
<?php

declare(strict_types=1);

namespace SolucionPhp\Shop\Notifier;

use SolucionPhp\Shop\MiniShop;

class ShopSmsNotifier implements NotifierInterface
{
    //private SmsNotifier $notifier;

    public function __construct()
    {
        //$this->notifier = new Notifier();
    }

    public function send(MiniShop $shop): void
    {
        //$text = sprintf("Your shop %s has been created", $shop->getName());

        try {
            //$this->notifier->send($text, $shop->getPhone());
        } catch (\Exception $e) {
            // Log this error
        }
    }
}

File: SRP/Kata_ Shop/SolucionJava/Shop/Notifier/ShopSmsNotifier.java
package SolucionJava.Shop.Notifier;

public class ShopSmsNotifier implements NotifierInterface {
    //SmsNotifier notifier;

    public ShopSmsNotifier() {
        //this.notifier = new SmsNotifier();
    }

    public void send(MiniShop shop) throws Exception {
        try {
            //String text = String.format("Your shop %s has been created", shop.getName());
            //this.notifier.send(text, shop.getPhone());
        } catch (Exception e) {
            // Log this fail
            throw new Exception("Something like wrong when trying to send notification");
        }
    }
}
